### PR TITLE
✨ GitLab: Code Review check

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Adds GitLab support for the Code-Review check.

#### What is the new behavior (if this is a feature change)?**
Code reviews in GitLab, including implicit reviews and comments, are now detected by Scorecard. This check is still slow because of the number of API calls per commit.

- [x] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?
```release-note
GitLab support for Code-Review check
```

### CODE REVIEW

File: clients/gitlabrepo/commits.go [hunk 1]
```diff
@@ -92,13 +68,6 @@ func (handler *commitsHandler) setup() error {
 						mergeRequest = mergeRequests[i]
```
Comment (azeemshaikh38) [inline clients/gitlabrepo/commits.go:68]: This line will seg fault. `mergeRequest` is guaranteed to be `nil` when it reaches this line for the first time.

Comment (azeemshaikh38) [inline clients/gitlabrepo/commits.go:68]: Remove unecessary `if len(mergeRequests) > 0`. Can simply do:

```
var mergeRequest *gitlab.MergeRequest
for i := range mergeRequests {
  ...
}
```

File: clients/gitlabrepo/commits.go [hunk 2]
```diff
@@ -107,15 +76,42 @@ func (handler *commitsHandler) setup() error {
...
+				if reviewer.State == "unreviewed" {
+					continue
+				}
+				reviews[reviewer.ID] = clients.Review{
+					Author: &clients.User{Login: reviewer.Username, ID: int64(reviewer.ID)},
+					State:  "COMMENTED",
+				}
+			}
+
+			approvals, _, err := handler.glClient.MergeRequests.GetMergeRequestApprovals(
```
Comment (azeemshaikh38) [inline clients/gitlabrepo/commits.go:95]: Can we avoid this extra API call by using GitLab's GraphQL API - https://docs.gitlab.com/ee/api/graphql/reference/#mergerequest ?

**Review by azeemshaikh38 — COMMENTED** [review]
Overall looks good. Left a comment about optimizing the API calls a bit more with GitLab graphQL.

File: clients/gitlabrepo/commits.go (hunk 1 above)
Comment (raghavkaul) [reply clients/gitlabrepo/commits.go:68]: I believe it's set on L62: `mergeRequest = mergeRequests[0]`. Or do you mean that's always nil?

Comment (azeemshaikh38) [reply clients/gitlabrepo/commits.go:68]: The first time this line is reached when all `mergeRequests[0:i-1]` are `nil`. So `mergeRequests[0] == mergeRequest == nil`.

**Review by azeemshaikh38 — APPROVED** [review]

**Pull request merged**